backend/tests/test_error_handling.py

```
@@ -0,0 +1,306 @@
+        assert "error" in error_data
+        assert error_data["error"]["code"] == "VALIDATION_ERROR"
+        assert "correlation_id" in error_data["error"]
+        assert "timestamp" in error_data["error"]
+        assert "details" in error_data["error"]
+    
+    def test_resource_not_found_error(self):
+        """Test resource not found error handling."""
+        with patch.object(integration_service, 'get_integration', side_effect=ResourceNotFoundError("Integration", "test-id")):
+            response = self.client.get(
+                "/v1/integrations/non-existent-id",
+                headers={"Authorization": f"Bearer test-token"}
```

Ruff (0.14.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)